src/main/java/com/google/devtools/build/lib/remote/TracingMetadataUtils.java

```diff
@@ -97,13 +112,14 @@ public static ClientInterceptor attachMetadataFromContextInterceptor() {
     @Override
     public <ReqT, RespT> Listener<ReqT> interceptCall(
         ServerCall<ReqT, RespT> call, Metadata headers, ServerCallHandler<ReqT, RespT> next) {
-      RequestMetadata meta = headers.get(METADATA_KEY);
-      if (meta == null) {
+      RequestMetadata meta;
+      try {
+        meta = extractRequestMetadata(headers);
+      } catch (IllegalStateException e) {
```

Comment:
nit: why translate one IllegalStateException to another? Just remove the try/catch

Reply:
My thought was to give a higher level cause for this particular error. On second thought, I will remove it though. Done.